### ISSUE TITLE
Difference in the number of initialised modules on running terraform init

### ISSUE THREAD

**Issue by rohanpaithankar-theiconic** [terraform-google-modules/terraform-google-bootstrap#190]
### TL;DR

I ran the terraform init one week ago and it downloaded all the dependent modules. A few days later, there are fewer modules being downloaded. No changes have been made to the scripts.

### Expected behavior

These are the old logs for modules expected to be downloaded on running `terraform init` command:

```
Already have image (with digest): {container registry path}
./verify_dependencies.sh
+ TERRAFORM_VERSION=v1.0.7
+ grep 'Terraform v1.0.7'
+ terraform version
Terraform v1.0.7
cp providers.template.tf  components/bootstrap/providers.tf
cd components/bootstrap && terraform init
Initializing modules...
Downloading terraform-google-modules/group/google 0.4.0 for billing_admins...
- billing_admins in .terraform/modules/billing_admins
- bootstrap in ../../modules/bootstrap
Downloading terraform-google-modules/bootstrap/google 3.1.0 for bootstrap.cloudbuild_bootstrap...
- bootstrap.cloudbuild_bootstrap in .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild
Downloading terraform-google-modules/project-factory/google 10.1.1 for bootstrap.cloudbuild_bootstrap.cloudbuild_project...
- bootstrap.cloudbuild_bootstrap.cloudbuild_project in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.budget in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/budget
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.gsuite_group in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/gsuite_group
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.project-factory in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/core_project_factory
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.project-factory.project_services in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/project_services
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.shared_vpc_access in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/shared_vpc_access
Downloading terraform-google-modules/bootstrap/google 3.1.0 for bootstrap.seed_bootstrap...
- bootstrap.seed_bootstrap in .terraform/modules/bootstrap.seed_bootstrap
Downloading terraform-google-modules/kms/google 1.2.0 for bootstrap.seed_bootstrap.kms...
- bootstrap.seed_bootstrap.kms in .terraform/modules/bootstrap.seed_bootstrap.kms
Downloading terraform-google-modules/project-factory/google 10.1.1 for bootstrap.seed_bootstrap.seed_project...
- bootstrap.seed_bootstrap.seed_project in .terraform/modules/bootstrap.seed_bootstrap.seed_project
- bootstrap.seed_bootstrap.seed_project.budget in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/budget
- bootstrap.seed_bootstrap.seed_project.gsuite_group in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/gsuite_group
- bootstrap.seed_bootstrap.seed_project.project-factory in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/core_project_factory
- bootstrap.seed_bootstrap.seed_project.project-factory.project_services in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/project_services
- bootstrap.seed_bootstrap.seed_project.shared_vpc_access in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/shared_vpc_access
Downloading terraform-google-modules/group/google 0.4.0 for organization_admins...
- organization_admins in .terraform/modules/organization_admins
Downloading terraform-google-modules/group/google 0.4.0 for security_admins...
- security_admins in .terraform/modules/security_admins

Initializing the backend...
```

### Observed behavior

Some root modules are not being downloaded, leading to 'Unsupported argument' errors in child modules on running `terraform plan` command. Following are the logs for modules being downloaded currently while runing the `terraform init` command.

```
Already have image (with digest): {container registry path}
./verify_dependencies.sh
+ TERRAFORM_VERSION=v1.0.7
+ terraform version
+ grep 'Terraform v1.0.7'
Terraform v1.0.7
cp providers.template.tf  components/bootstrap/providers.tf
cd components/bootstrap && terraform init
Initializing modules...
- bootstrap in ../../modules/bootstrap
Downloading terraform-google-modules/bootstrap/google 3.1.0 for bootstrap.cloudbuild_bootstrap...
- bootstrap.cloudbuild_bootstrap in .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild
Downloading terraform-google-modules/project-factory/google 10.1.1 for bootstrap.cloudbuild_bootstrap.cloudbuild_project...
- bootstrap.cloudbuild_bootstrap.cloudbuild_project in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project
Downloading terraform-google-modules/bootstrap/google 3.1.0 for bootstrap.seed_bootstrap...
- bootstrap.seed_bootstrap in .terraform/modules/bootstrap.seed_bootstrap
Downloading terraform-google-modules/kms/google 1.2.0 for bootstrap.seed_bootstrap.kms...
- bootstrap.seed_bootstrap.kms in .terraform/modules/bootstrap.seed_bootstrap.kms
Downloading terraform-google-modules/project-factory/google 10.1.1 for bootstrap.seed_bootstrap.seed_project...
- bootstrap.seed_bootstrap.seed_project in .terraform/modules/bootstrap.seed_bootstrap.seed_project
- Downloading terraform-google-modules/group/google 0.4.0 for organization_admins...
- organization_admins in .terraform/modules/organization_admins
Downloading terraform-google-modules/group/google 0.4.0 for security_admins...
- security_admins in .terraform/modules/security_admins

Initializing the backend...
```

The following logs are not present:
```
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.budget in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/budget
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.gsuite_group in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/gsuite_group
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.project-factory in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/core_project_factory
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.project-factory.project_services in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/project_services
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.shared_vpc_access in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/shared_vpc_access
- bootstrap.seed_bootstrap.seed_project.budget in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/budget
- bootstrap.seed_bootstrap.seed_project.gsuite_group in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/gsuite_group
- bootstrap.seed_bootstrap.seed_project.project-factory in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/core_project_factory
- bootstrap.seed_bootstrap.seed_project.project-factory.project_services in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/project_services
- bootstrap.seed_bootstrap.seed_project.shared_vpc_access in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/shared_vpc_access
```

### Terraform Configuration

```hcl
terraform {
  required_version = ">= 0.13"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = ">= 3.50"
    }
  }
}
```

### Terraform Version

```sh
Terraform v1.0.7
```

**Comment by rohanpaithankar-theiconic** [comment]
Raised the issue in the [terraform-example-foundation](https://github.com/terraform-google-modules/terraform-example-foundation) repository.
